### PR TITLE
feat: add v-client:visible directive

### LINKED ISSUE #72: feat: add v-client:visible directive
## Summary

Add a `v-client:visible` directive that defers island hydration until the component enters the viewport using `IntersectionObserver`.

## Motivation

Currently, `v-client:load` is the only client directive, which hydrates all islands immediately on page load. For pages with many islands or below-the-fold interactive components, this is wasteful. `v-client:visible` would allow deferring hydration until the user actually scrolls to the component.

This is equivalent to Astro's [`client:visible`](https://docs.astro.build/en/reference/directives-reference/#clientvisible) directive.

## Usage

```vue
<HeavyCarousel v-client:visible />
```

## Implementation Notes

- Extend `findVClientElements()` in `src/build/sfc-analysis.ts` to recognize `visible` argument
- Pass a `strategy="visible"` attribute on the generated `<vue-island>` element in `src/build/generate.ts`
- In `src/client/custom-element.ts`, use `IntersectionObserver` to defer hydration in `connectedCallback()` until the element is visible
- Consider supporting a `rootMargin` option for pre-loading (Astro supports this)

### PR DESCRIPTION
**Pull request by ktsn**
[Posted by an agent]

Closes #72

## Summary
- Extended `findVClientElements()` in `src/build/sfc-analysis.ts` to recognize both `load` and `visible` arguments, returning a `VClientMatch` object with `element` and `strategy` fields
- Added `parseIslandWrapId()` / `buildIslandWrapId()` helpers in `src/build/generate.ts` to encode the strategy in the island wrapper virtual module ID (e.g. `\0visle:island-wrap:visible:/path/to/Comp.vue`)
- Updated `generateIslandWrapperCode()` to emit a `strategy` attribute on the `<vue-island>` element when the strategy is not `load`
- Updated `src/build/plugins/server-transform.ts` to thread the strategy through the island import map and virtual module resolution
- In `src/client/custom-element.ts`, the `connectedCallback()` now checks for `strategy="visible"` and uses `IntersectionObserver` to defer hydration until the element enters the viewport
- Supports an optional `root-margin` attribute for pre-loading (passed to `IntersectionObserver` as `rootMargin`)
- Added tests for the `visible` strategy in both `sfc-analysis.test.ts` and `custom-element.test.ts`

## Summary by CodeRabbit

* **New Features**
  * Islands now support selectable hydration strategies: immediate load or visibility-based (hydrates when an element becomes visible). Visibility-based hydration respects a configurable root-margin.
  * Strategy information is propagated through the build/runtime so directives with a strategy are handled consistently.

* **Tests**
  * Added comprehensive tests covering visibility-based hydration, mixed strategies, and related edge cases.

### CODE REVIEW

File: src/build/plugins/server-transform.ts [hunk 1]
```diff
@@ -193,12 +205,12 @@ export function serverTransformPlugin(): ServerTransformPluginResult {
...
-          set = new Set()
-          islandImportMap.set(fileName, set)
+        let map = islandImportMap.get(fileName)
+        if (!map) {
+          map = new Map()
+          islandImportMap.set(fileName, map)
         }
-        set.add(resolvedPath)
+        map.set(resolvedPath, strategy)
       }
```
[the remark this answers is not in the document]
Comment (ktsn) [reply src/build/plugins/server-transform.ts:220]: [Posted by an agent]

Good catch. Added a conflict guard that emits a build error when the same component is used with different v-client strategies in the same file. Pushed in 5753da7.
